cpp/include/cudf/strings/strings_column_handler.hpp

```diff
@@ -0,0 +1,232 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <rmm/thrust_rmm_allocator.h>
+#include <cudf/column/column_view.hpp>
+#include <cudf/column/column.hpp>
+
+namespace cudf {
+
+/**---------------------------------------------------------------------------*
+ * @brief Given a column-view of strings type, an instance of this class
+ * provides the strings operations on the column.
+ *---------------------------------------------------------------------------**/
+class strings_column_handler
+{
+ public:
+  ~strings_column_handler() = default;
+
+  strings_column_handler( column_view strings_column,
+                          rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource() );
+  //strings_column_handler( const column_view&& strings_column );
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the number of strings in the column
+   *---------------------------------------------------------------------------**/
+  size_type size() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the internal parent string column
+   *---------------------------------------------------------------------------**/
+  column_view parent_column() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the internal column of offsets
+   *---------------------------------------------------------------------------**/
+  column_view offsets_column() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the internal column of chars
+   *---------------------------------------------------------------------------**/
+  column_view chars_column() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns a pointer to the internal null mask memory
+   *---------------------------------------------------------------------------**/
+  const bitmask_type* null_mask() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the number of nulls in this column
+   *---------------------------------------------------------------------------**/
+  size_type null_count() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the registered memory resource
+   *---------------------------------------------------------------------------**/
+  rmm::mr::device_memory_resource* memory_resource() const;
```

Comment:
This doesn't need a memory resource anymore, correct? The `mr` should be passed into the APIs like gather, sublist, etc.